src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -269,10 +271,104 @@ public void Stop()
+            byte[] pendingDocIds = null;
+            pendingDocIds = Native.c4repl_getPendingDocIDs(_repl, &err);
+
+            if (err.code > 0) {
+                CBDebug.LogAndThrow(WriteLog.To.Sync, CouchbaseException.Create(err), Tag, err.ToString(), true);
+            }
+
+            if (pendingDocIds != null) {
+                _databaseThreadSafety.DoLocked(() => {
+                    var flval = Native.FLValue_FromData(pendingDocIds, FLTrust.Trusted);
+                    var flarr = Native.FLValue_AsArray(flval);
+                    var cnt = (int) Native.FLArray_Count(flarr);
+                    for (int i = 0; i < cnt; i++) {
+                        var flv = Native.FLArray_Get(flarr, (uint) i);
+                        result.Add(Native.FLValue_AsString(flv));
+                    }
+
+                    Array.Clear(pendingDocIds, 0, pendingDocIds.Length);
+                    pendingDocIds = null;
+                });
+            }
+
+            _pendingDocIds = result.ToImmutableHashSet<string>();
+            return _pendingDocIds;
+        }
+
+        /// <summary>
+        /// Checks whether or not a document with the given ID has any pending revisions to push
+        /// </summary>
+        /// <param name="documentID">The document ID</param>
+        /// <returns>A bool which represents whether or not the document with the corresponding ID has one or more pending revisions.  
+        /// <c>true</c> means that one or more revisions have not been pushed to the remote yet, 
+        /// and <c>false</c> means that all revisions on the document have been pushed</returns>
+        /// <exception cref="CouchbaseLiteException">Thrown if no push replication</exception>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        public bool IsDocumentPending([NotNull]string documentID)
+        {
+            CBDebug.MustNotBeNull(WriteLog.To.Sync, Tag, nameof(documentID), documentID);
+            bool isDocPending = false;
+
+            if (!IsPushing()) {
+                CBDebug.LogAndThrow(WriteLog.To.Sync,
+                    new CouchbaseLiteException(C4ErrorCode.Unsupported, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs),
+                    Tag, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs, true);
+            }
+
+            DispatchQueue.DispatchSync(() => {
+                var errSetupRepl = SetupC4Replicator();
+                if (errSetupRepl.code > 0) {
+                    CBDebug.LogAndThrow(WriteLog.To.Sync, CouchbaseException.Create(errSetupRepl), Tag, errSetupRepl.ToString(), true);
+                }
+            });
+
+            C4Error err = new C4Error();
+            isDocPending = Native.c4repl_isDocumentPending(_repl, documentID, &err);
+            if (err.code > 0) {
+                CBDebug.LogAndThrow(WriteLog.To.Sync, CouchbaseException.Create(err), Tag, err.ToString(), true);
```

Comment:
This is what `LiteCoreBridge` is for (and above as well)

Reply:
Looks like LiteCoreBridge.Check doesn't get me what I need.. In the pending doc ids, I need retrieve byte[] as pending doc ids (converted from FLSlice in C4Replicator_native) and bool as is document pending..

Reply:
It might not be set up to handle `byte[]` but it can be modified to do so if desired.  It already handles bools though via this delegate signature https://github.com/couchbase/couchbase-lite-net/blob/26e597cc00a9245a3ace9b7ff39da1994b84443e/src/LiteCore/src/LiteCore.Shared/Interop/NativeHandler.cs#L30-L34